### ISSUE TITLE
AngularFireUploadTask snapshot state doesn't change upon cancellation - AngularFireStorage

### ISSUE THREAD

**Issue by yuliankarapetkov** [angular/angularfire#1805]
### Version info

**Angular:** 6.03

**Firebase:** 5.2.0

**AngularFire:** 5.0.0-rc.11

### How to reproduce these conditions

**Steps to set up and reproduce**

Subscribe to `snapshotChanges` and log the value of the snapshot. Cancel the `AngularFireUploadTask` through the `cancel()` method. Snapshot doesn't change => `snapshot.state === 'running'`, instead of `snapshot.state === 'cancelled'`

### Expected behavior

`snapshot` has a new value with `state === 'cancelled'`

### Actual behavior

`snapshot` doesn't have a new value, so `state === 'running'`

**Comment by davideast** [comment]
@yuliankarapetkov Can you create a stackblitz for me to debug?

**Comment by yuliankarapetkov** [comment]
@davideast Hey, thanks for looking into this for me. 

Here's a [stackblitz link](https://stackblitz.com/edit/angularfire2-cancel-upload)

**Comment by yuliankarapetkov** [comment]
@davideast did you get the chance to check the stackblitz link provided?

**Comment by alexdabast** [comment]
Seems to be a real bug
The last event is not emitted on the snapshotChanges() however it will be emitted in the .then() promise



```
this._task.snapshotChanges()
      .subscribe(snapshot => {
        this.snapshot = snapshot; // will be emitted during progress and cancel
      });
    this._task.then((snapshot) => {
      this.snapshot = snapshot; // will be emitted when upload is done
    });
```

**Comment by jamesdaniels** [comment]
Both of these issues should have been addressed in a recent 6.0 patch, we were not updating the taskSnapshot on completion just completing the observable. Similarly I realized error wasn't updating the snapshot and have made a PR for that #2665 which will be in 6.1.1
